This notebook was created by Skylar Miller and Ellis Hartley

# Assumptions

- Domestic company only interested in domestic gross revenue (for seasonality consistency)
- 


# Question 

Does the season of release affect the revenue of a movie?

# Backround 

We think that the summer months will perform the best because students are out of school and families will be looking for ways to entertain. In addition, with heat waves in full effect, in activities are preferred. 

### Null Hypothesis: There is no difference between summer month revenue and other month revenues. 
 
### Alternative Hypothesis: There is more summer month revenue than other month revenues.

# Data Exploration
- Opening all of the available data sources, we decided to go with The Numbers Dataset. More info on this dataset can be found here: https://www.the-numbers.com/

In [6]:
import pandas as pd
df_bom = pd.read_csv('../../zippedData/bom.movie_gross.csv.gz')
df_bom.head(1)

title studio  domestic_gross foreign_gross  year
0  Toy Story 3     BV     415000000.0     652000000  2010

In [7]:
df_tmbd = pd.read_csv('../../zippedData/tmdb.movies.csv.gz')
df_tmbd.head(1)

Unnamed: 0        genre_ids     id original_language  \
0           0  [12, 14, 10751]  12444                en   

                                 original_title  popularity release_date  \
0  Harry Potter and the Deathly Hallows: Part 1      33.533   2010-11-19   

                                          title  vote_average  vote_count  
0  Harry Potter and the Deathly Hallows: Part 1           7.7       10788

In [8]:
df_tn = pd.read_csv('../../zippedData/tn.movie_budgets.csv.gz')
df_tn.head(1)

id  release_date   movie production_budget domestic_gross worldwide_gross
0   1  Dec 18, 2009  Avatar      $425,000,000   $760,507,625  $2,776,345,279

In [9]:
df_rt_info = pd.read_csv('../../zippedData/rt.movie_info.tsv.gz', sep='\t')
df_rt_info.head(1)

id                                           synopsis rating  \
0   1  This gritty, fast-paced, and innovative police...      R   

                                 genre          director          writer  \
0  Action and Adventure|Classics|Drama  William Friedkin  Ernest Tidyman   

  theater_date      dvd_date currency box_office      runtime studio  
0  Oct 9, 1971  Sep 25, 2001      NaN        NaN  104 minutes    NaN

In [10]:
df_rt_reviews = pd.read_csv('../../zippedData/rt.reviews.tsv.gz', sep='\t', encoding='ansi')
df_rt_reviews.head(1)

id                                             review rating  fresh  \
0   3  A distinctly gallows take on contemporary fina...    3/5  fresh   

       critic  top_critic        publisher               date  
0  PJ Nabarro           0  Patrick Nabarro  November 10, 2018

In [11]:
# summer = june 21 - sept 22 (06/21-09/22) # summer months = 6,7,8
# fall = sept 23 - dec 20 (09/23-12/20) # fall months = 9,10,11
# winter = dec 21 - mar 18 (12/21-03/18) # winter months = 12,1,2
# spring = mar 19 - jun 20 (03/19-06/20) # spring months = 3,4,5

# spring = df_tn[(df_tn['release_date'].str.startswith('Mar'))|
#                (df_tn['release_date'].str.startswith('Apr'))|
#                (df_tn['release_date'].str.startswith('May'))]
# spring

In [12]:
df_tn.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5782 entries, 0 to 5781
Data columns (total 6 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   id                 5782 non-null   int64 
 1   release_date       5782 non-null   object
 2   movie              5782 non-null   object
 3   production_budget  5782 non-null   object
 4   domestic_gross     5782 non-null   object
 5   worldwide_gross    5782 non-null   object
dtypes: int64(1), object(5)
memory usage: 271.2+ KB


As shown above, this dataset consists of 5782 rows with no Null values. 

In [13]:
def clean_dollar(df, column):
    df[column] = df[column].str.strip('$')
    df[column] = df[column].str.replace(',','')
    return df[column].astype(int)

clean_dollar(df_tn,'production_budget') 
clean_dollar(df_tn,'domestic_gross')
# clean_dollar(df_tn,'worldwide_gross')

0       760507625
1       241063875
2        42762350
3       459005868
4       620181382
          ...    
5777            0
5778        48482
5779         1338
5780            0
5781       181041
Name: domestic_gross, Length: 5782, dtype: int32

In [14]:
df_tn.head()

id  release_date                                        movie  \
0   1  Dec 18, 2009                                       Avatar   
1   2  May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
2   3   Jun 7, 2019                                 Dark Phoenix   
3   4   May 1, 2015                      Avengers: Age of Ultron   
4   5  Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   

  production_budget domestic_gross worldwide_gross  
0         425000000      760507625  $2,776,345,279  
1         410600000      241063875  $1,045,663,875  
2         350000000       42762350    $149,762,350  
3         330600000      459005868  $1,403,013,963  
4         317000000      620181382  $1,316,721,747